packages/typespec-azure-core/test/rules/use-standard-integer.test.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  BasicTestRunner,
+  LinterRuleTester,
+  createLinterRuleTester,
+} from "@typespec/compiler/testing";
+import { beforeEach, it } from "vitest";
+import { useStandardInteger } from "../../src/rules/use-standard-integer.js";
+import { createAzureCoreTestRunner } from "../test-host.js";
+
+let runner: BasicTestRunner;
+let tester: LinterRuleTester;
+
+beforeEach(async () => {
+  runner = await createAzureCoreTestRunner({ omitServiceNamespace: true });
+  tester = createLinterRuleTester(runner, useStandardInteger, "@azure-tools/typespec-azure-core");
+});
+
+it("emits a warning diagnostic for non-standard integer types", async () => {
+  await tester
+    .expect(
+      `
+      namespace Azure.Widget;
+
+      model Widget {
+        prop1: uint16;
+        prop2: int8;
+        prop3: uint32;
+      }
+      `
+    )
+    .toEmitDiagnostics([
+      {
+        code: "@azure-tools/typespec-azure-core/use-standard-integer",
+      },
+      {
+        code: "@azure-tools/typespec-azure-core/use-standard-integer",
+      },
+      {
+        code: "@azure-tools/typespec-azure-core/use-standard-integer",
+      },
+    ]);
+});
+
+it("does not emit a warning diagnostic for standard integer types", async () => {
+  await tester
+    .expect(
+      `
+      namespace Azure.Widget;
+
+      model Widget {
+        prop1: int32;
+        prop2: int64;
+        prop3: safeint;
+      }
+      `
+    )
+    .toBeValid();
+});
+
+it("does not emit a warning diagnostic for non-standard integer types that map to supported integers in Autorest", async () => {
```

Comment:
Do we want this test?  I think both numeric and integer should produce a separate warning about imprecise numeric types

Reply:
I think so, but I could just move these into the above test. I'm not testing for the non-precise integer diagnostic (there's already a test for that) but just to make sure that the "use-standard-int" is not thrown for these since they make to "int64" in autorest.

Reply:
These tests were refactored